File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        
        ListNode* temp =head;
        int len=0;
        while(temp)
        {
            len++;
            temp=temp->next;
        }
        int remove=len-n;
        // cout<<remove<<" ";
        if (len==1 && n==1 )
        {
            return nullptr;
        }
        if(remove==0)
        {
            return head->next;
        }
        temp=head;
        int pos=0;
        while(pos<remove-1 &&temp->next)
        {
            temp=temp->next;
            pos++;
        }
        temp->next=temp->next->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head==null || head.next==null){
            return null;
        }
        ListNode fwd= head;

        while(--n>0){
            fwd= fwd.next;
        }
        
        ListNode bkd= head;

        ListNode prv= head;
        
        while(fwd.next != null){
            prv= bkd;
            fwd= fwd.next;
            bkd= bkd.next;
        }
        if(bkd==head)
        {
            return head.next; 
        }
        
        // if(cur != null && prv != null)
            prv.next= bkd.next;
        // else{
        //     prv= null;
        // }
        // if(prv==head)
        // {
        //     return head.next;
        // }
        return head;
    }
    }